test/DAppIntegration.t.sol

```diff
@@ -0,0 +1,154 @@
+import { FastLaneErrorsEvents } from "../src/contracts/types/Emissions.sol";
+
+import { DummyDAppControl } from "./base/DummyDAppControl.sol";
+
+import "../src/contracts/types/GovernanceTypes.sol";
+
+contract MockDAppIntegration is DAppIntegration {
+    constructor(address _atlas) DAppIntegration(_atlas) { }
+
+    function initializeNonceInternal(address account) external returns (bool) {
+        return _initializeNonce(account);
+    }
+}
+
+contract DAppIntegrationTest is Test {
+    MockDAppIntegration public dAppIntegration;
+    DummyDAppControl public dAppControl;
+
+    address governance = makeAddr("governance");
+    address signatory = makeAddr("signatory");
+
+    function setUp() public {
+        dAppIntegration = new MockDAppIntegration(address(0));
+        dAppControl = new DummyDAppControl(address(0), governance);
+    }
+
+    function test_initializeGovernance() public {
+        vm.expectRevert(FastLaneErrorsEvents.OnlyGovernance.selector);
+        dAppIntegration.initializeGovernance(address(dAppControl));
+
+        bytes32 signatoryKey = keccak256(abi.encode(governance, address(dAppControl)));
+        assertFalse(dAppIntegration.signatories(signatoryKey));
+
+        vm.prank(governance);
+        dAppIntegration.initializeGovernance(address(dAppControl));
+
+        assertTrue(dAppIntegration.signatories(signatoryKey));
+
+        vm.prank(governance);
+        vm.expectRevert(FastLaneErrorsEvents.OwnerActive.selector);
+        dAppIntegration.initializeGovernance(address(dAppControl));
+    }
+
+    function test_addSignatory() public {
+        vm.prank(governance);
+        dAppIntegration.initializeGovernance(address(dAppControl));
+
+        vm.expectRevert(FastLaneErrorsEvents.OnlyGovernance.selector);
+        dAppIntegration.addSignatory(address(dAppControl), signatory);
+
+        bytes32 signatoryKey = keccak256(abi.encode(governance, signatory));
+        assertFalse(dAppIntegration.signatories(signatoryKey));
+
+        vm.prank(governance);
+        dAppIntegration.addSignatory(address(dAppControl), signatory);
+
+        assertTrue(dAppIntegration.signatories(signatoryKey));
+
+        vm.prank(governance);
+        vm.expectRevert(FastLaneErrorsEvents.SignatoryActive.selector);
+        dAppIntegration.addSignatory(address(dAppControl), signatory);
+    }
+
+    function test_removeSignatory() public {
+        vm.startPrank(governance);
+        dAppIntegration.initializeGovernance(address(dAppControl));
+        dAppIntegration.addSignatory(address(dAppControl), signatory);
+        vm.stopPrank();
+
+        vm.expectRevert(FastLaneErrorsEvents.InvalidCaller.selector);
+        dAppIntegration.removeSignatory(address(dAppControl), signatory);
+
+        bytes32 signatoryKey = keccak256(abi.encode(governance, signatory));
+        assertTrue(dAppIntegration.signatories(signatoryKey));
+
+        vm.prank(governance);
+        dAppIntegration.removeSignatory(address(dAppControl), signatory);
```

Comment:
This test doesn't seem to test that `signatory` can remove itself as well, which I think is what this part is getting at:

```
        if (msg.sender != govData.governance && msg.sender != signatory) {
            revert FastLaneErrorsEvents.InvalidCaller();
        }
```

Reply:
Added in https://github.com/FastLane-Labs/atlas/pull/87/commits/ce6e8d9e758d2905ff9fd39e9df0d5934f0ab220.

Nice catch, it actually uncover a mistake in `DAppIntegration.removeSignatory()`, the `signatoryKey` was not properly generated. Thanks!